### PR TITLE
Adds veganism disability

### PR DESCRIPTION
**Pull request by unid15**
See changelog. Players who become vampires lose the veganism gene, but it can be activated again through genetics.
Consuming animal products as a vegan causes you to become nauseous. If you continue eating them while nauseous, you receive 2.5 toxin damage per bite (in case of food), or per 10 ticks (in case of reagents such as milk or honey). Yes, this means you can murder vegans by force-feeding them steaks.

Also adds vegan syndrome as a stage 2 virus effect, and a veganism/animal rights religion

:cl:
 * rscadd: Added a new disability - veganism. It can be received either randomly at roundstart, by enabling it in your character preferences, through a new stage 2 virus effect, or through genetics. Also added a "veganism" / "animal rights" religion which automatically grants this disability.
 * rscadd: Eating animal products, meat, milk, honey and blood as a vegan will make you nauseous, and further continuing to do so will cause minor toxin damage. Vegan vampires are unable to drink blood, if turned vegan after becoming a vampire.

### CODE REVIEW

**Comment by jknpj** [comment]
perfect, just needs to be a preference too

**Comment by N3X15** [comment]
DOESN'T HAVE <s>ITS OWN RELIGION AND</s> BRAINDAMAGE-LIKE YELLING, DOWNBOATED

**Comment by ShiftyRail** [comment]
@N3X15 
>DOESN'T HAVE ITS OWN RELIGION

Yet !
https://github.com/vgstation-coders/vgstation13/pull/16465

@unid15 
This is going to conflict my PR on religions datums though.

**Comment by Aurx** [comment]
Needs to give them highly pretentious messages when they do eat vegan food.

File: code/game/dna/genes/vg_disabilities.dm [hunk 1]
```diff
@@ -69,3 +69,46 @@
...
+
+/datum/dna/gene/disability/veganism
+	name = "Veganism"
+	desc = "Causes the digestive system to completely reject all animal products, from meat to dairy."
+	activation_message = "You feel vegan."
+	deactivation_message = "You're back on top of the food chain."
+
+	mutation = M_VEGAN
+
+	var/static/list/bad_reagents = list(
```
Comment (ghost) [inline code/game/dna/genes/vg_disabilities.dm]: Can you really not make this a variable at the reagent level

Comment (unid15) [reply code/game/dna/genes/vg_disabilities.dm]: Done, now a global variable

**Comment by unid15** [comment]
- Added a stage 2 virus effect named "Vegan Syndrome" that makes the affected mob vegan
- Veganism can be enabled in character preferences

**Comment by Blithering** [comment]
Can you make vegan vampires drink tomato juice?

**Comment by unid15** [comment]
There's nothing preventing them from doing it atm, but if you mean giving them blood/powers for drinking tomato juice - that'd be overpowered

**Comment by Blithering** [comment]
even if its at a really low exchange rate? completly cockblocking an antag type with a single disability is a bit harsh

**Comment by unid15** [comment]
Players lose veganism when they're turned into vampires, so you can't accidentally become a vegan vampire. One way vampire veganism could be abused is by security forcefully turning every crewmember vegan when there's a rumor of vampires, but that's 1) funny 2) not very different from (and in fact vastly inferior to) just spraying everybody with holy water.

**Comment by FudgePucker** [comment]
Make a virus that does it.
Arrest anybody who tries to get rye.

**Comment by Dacendeth** [comment]
What if you're a vegan and some madman has put milk in cryo for _some reason_?

**Comment by Exxion** [comment]
Then you'd best hope they also put rye in, it seems

**Comment by Dacendeth** [comment]
Ohh it's a genetic thing too I missed that, does that mean the human/monkey genetic block number is changing since there will be one more thing?

**Comment by unid15** [comment]
It's always the last block

**Pull request merged**